pandas/_config/config.py

```diff
@@ -455,10 +453,12 @@ def register_option(key, defval, doc="", validator=None, cb=None):
     path = key.split(".")
 
     for k in path:
-        if not bool(re.match("^" + tokenize.Name + "$", k)):
-            raise ValueError("{k} is not a valid identifier".format(k=k))
+        # NOTE: tokenize.Name is not a public constant
+        # error: Module has no attribute "Name"  [attr-defined]
+        if not re.match("^" + tokenize.Name + "$", k):  # type: ignore
```

Comment:
Maybe worth considering `tokenize.Name` to `\w+`; it hasn't changed in 9 years:

https://github.com/python/cpython/blob/305189ecdc8322c22879a04564cad5989f937462/Lib/tokenize.py#L67